components/fireside/post/like/widget/widget.ts

```diff
@@ -57,6 +89,13 @@ export default class AppFiresidePostLikeWidget extends Vue {
 	async toggleLike() {
 		const currentLike = this.post.user_like;
 		if (!currentLike) {
+			// Do this before attempting to follow.
+			// We don't want to wait till the follow is confirmed to show the dialog,
+			// and even if the follow fails it's not like we'll close it.
+			if (this.shouldShowFollow && UserFollowSuggestion.canSuggest(this.post.user.id)) {
```

Comment:
all of this code is duplicated. should it be? can we reduce it somehow? put it in one place or something

Reply:
I didn't want to get into refactoring existing code too much. this was supposed to be a ~1 hour tops feature, mind.

We could probably deduplicate the follow widgets in some slot based component that does the follow suggestion but the savings arent worth the refactor time so meh.